src/test/java/redis/clients/jedis/tests/commands/StreamsCommandsTest.java

```diff
@@ -306,6 +319,166 @@ public void xpendeing() {
     jedis.xclaim("xpendeing-stream", "xpendeing-group", "xpendeing-consumer2", 500, 0, 0, false, pendingRange.get(0).getID());
   }
 
+  @Test
+  public void xinfo() {
+
+    final String STREAM_NAME = "xadd-stream1";
+    final String F1 = "f1";
+    final String V1 = "v1";
+    final String V2 = "v2";
+    final String G1 = "G1";
+    final String G2 = "G2";
+    final String MY_CONSUMER = "myConsumer";
+    final String MY_CONSUMER2 = "myConsumer2";
+
+    Map<String, String> map1 = new HashMap<>();
+    map1.put(F1, V1);
+    StreamEntryID id1 = jedis.xadd(STREAM_NAME, null, map1);
+    map1.put(F1, V2);
+    StreamEntryID id2 = jedis.xadd(STREAM_NAME, null, map1);
+    assertNotNull(id1);
+    StreamInfo streamInfo =jedis.xinfoStream(STREAM_NAME);
+    assertNotNull(id2);
+
+    jedis.xgroupCreate(STREAM_NAME,G1, StreamEntryID.LAST_ENTRY,false);
+    Entry<String, StreamEntryID> streamQeury11 = new AbstractMap.SimpleImmutableEntry<>(STREAM_NAME, new StreamEntryID("0-0"));
+    jedis.xreadGroup(G1, MY_CONSUMER,1,0,false,streamQeury11);
+    List<StreamGroupInfo> groupInfo = jedis.xinfoGroup(STREAM_NAME);
+    List<StreamConsumersInfo> consumersInfo = jedis.xinfoConsumers(STREAM_NAME, G1);
+
+    //Stream info test
+    assertEquals(2L,streamInfo.getStreamInfo().get(LENGTH));
+    assertEquals(1L,streamInfo.getStreamInfo().get(RADIX_TREE_KEYS));
+    assertEquals(2L,streamInfo.getStreamInfo().get(RADIX_TREE_NODES));
+    assertEquals(0L,streamInfo.getStreamInfo().get(GROUPS));
+    assertEquals(V1,((StreamEntry)streamInfo.getStreamInfo().get(FIRST_ENTRY)).getFields().get(F1));
+    assertEquals(V2,((StreamEntry)streamInfo.getStreamInfo().get(LAST_ENTRY)).getFields().get(F1));
+
+    //Using getters
+    assertEquals(2,streamInfo.getLength());
+    assertEquals(1,streamInfo.getRadixTreeKeys());
+    assertEquals(2,streamInfo.getRadixTreeNodes());
+    assertEquals(0,streamInfo.getGroups());
```

Comment:
Missing getLastGeneratedId()

Reply:
Done in a way that we check if it's not null - it is difficult to know what last-generated should be

Reply:
shouldn't it be id2?

Reply:
you are right. I've missed that we have that info already.
Maybe then I should change `getLastGeneratedId` method to return `StreamEntryID` instead of `String`?